airtime_mvc/application/forms/LiveStreamingPreferences.php

```diff
@@ -140,7 +153,53 @@ public function updateVariables()
 
     public function isValid($data)
     {
-        return parent::isValid($data);
+        $isValid = parent::isValid($data);
+        $master_source_port = $data['master_source_port'];
+        $show_source_port = $data['show_source_port'];
+
+        if ($master_source_port == $show_source_port && $master_source_port != "") {
+            $element = $this->getElement('show_source_port');
+            $element->addError(_("You cannot use same port as Master DJ port."));
+            $isValid = false;
+        }
+        if ($master_source_port != "") {
+            if (is_numeric($master_source_port)) {
+                if ($master_source_port != Application_Model_StreamSetting::getMasterLiveStreamPort()) {
+                    $sock = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
+                    try {
+                        socket_bind($sock, 0, $master_source_port);
+                    } catch (Exception $e) {
+                        $element = $this->getElement("master_source_port");
+                        $element->addError(sprintf(_("Port %s is not available"), $master_source_port));
+                        $isValid = false;
+                    }
+
+                    socket_close($sock);
+                }
+            } else {
+                $isValid = false;
+            }
+        }
+        if ($show_source_port != "") {
+            if (is_numeric($show_source_port)) {
+                if ($show_source_port != Application_Model_StreamSetting::getDjLiveStreamPort()) {
+                    $sock = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
+                    try {
+                        socket_bind($sock, 0, $show_source_port);
+                    } catch (Exception $e) {
+                        $element = $this->getElement("show_source_port");
+                        $element->addError(sprintf(_("Port %s is not available"), $show_source_port));
+                        $isValid = false;
+                    }
+                    socket_close($sock);
+                }
```

Comment:
Either with a tooltip or by calling `setDescription()` on the form element (like in with the CORS form). Also remember to wrap the string in the `_()` method call so it can be translated.